src/backend/distributed/deparser/deparse_schema_stmts.c

```diff
@@ -0,0 +1,105 @@
+/*-------------------------------------------------------------------------
+ *
+ * deparse_schema_stmts.c
+ *	  All routines to deparse schema statements.
+ *	  This file contains all entry points specific for type statement deparsing as well as
+ *	  functions that are currently only used for deparsing of the schema statements.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+#include "postgres.h"
+
+#include "distributed/citus_ruleutils.h"
+#include "distributed/deparser.h"
+#include "lib/stringinfo.h"
+#include "nodes/nodes.h"
+#include "utils/builtins.h"
+
+static void AppendGrantOnSchemaStmt(StringInfo buf, GrantStmt *stmt);
+
+char *
+DeparseGrantOnSchemaStmt(Node *node)
+{
+    GrantStmt *stmt = castNode(GrantStmt, node);
+    Assert(stmt->objtype == OBJECT_SCHEMA);
+
+    StringInfoData str = { 0 };
+    initStringInfo(&str);
+
+    AppendGrantOnSchemaStmt(&str, stmt);
+    
+    return str.data;
+}
+
+static void
+AppendGrantOnSchemaStmt(StringInfo buf, GrantStmt *stmt)
```

Comment:
The function body is relatively long. If you were to  split the for loops out to their own functions it would make the code smaller and easier readable.

 - `AppendPriviliges` (could contain the `NULL` check on `stmt-> privileges` as well)
 - `AppendSchemaNames`
 - `AppendGrantees`
 - etc

This will hide a lot of details by descriptive names which makes the code generally easier to follow